packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -378,14 +385,25 @@ contract Ethernauts is ERC721Enumerable, Ownable, ReentrancyGuard {
     function _mintNext(address to) private {
         uint256 tokenId = totalSupply();
 
-        uint256 currentBatchId = tokenId / batchSize;
-        uint256 maxTokenIdInBatch = batchSize * (currentBatchId + 1) - 1;
+        _generateRandomNumberIfNeeded();
+
+        _mint(to, tokenId);
+    }
+
+    function _generateRandomNumberIfNeeded() private {
+        uint256 lastTokenId = totalSupply();
```

Comment:
The FN `_generateRandomNumberIfNeeded` is only being called from `_mintNext`, and `_mintNext` already computes `uint256 tokenId = totalSupply();`. I know this way is clearer, but if we receive this value as param we will be saving gas on each mint.

Reply:
Perfect!